code/controllers/subsystem/mobs.dm

```diff
@@ -26,18 +27,54 @@ SUBSYSTEM_DEF(mobs)
 		dead_players_by_zlevel[dead_players_by_zlevel.len] = list()
 
 /datum/controller/subsystem/mobs/fire(resumed = FALSE)
-	if (!resumed)
-		src.currentrun = GLOB.mob_living_list.Copy()
+	if(!resumed)
+		currentrun = GLOB.mob_living_list.Copy()
 
-	//cache for sanic speed (lists are references anyways)
-	var/list/currentrun = src.currentrun
+	///Cache for speed as lists are references
+	var/list/currentprocrun = currentrun
 	var/times_fired = src.times_fired
-	while(currentrun.len)
-		var/mob/living/L = currentrun[currentrun.len]
-		currentrun.len--
-		if(L)
-			L.Life(times_fired)
-		else
-			GLOB.mob_living_list.Remove(L)
-		if (MC_TICK_CHECK)
+	while(currentprocrun.len)
+		var/mob/living/L = currentprocrun[currentprocrun.len]
+		currentprocrun.len--
+		if(!L.life_process())
+			GLOB.mob_living_list.Remove(L)//we died
+
+		if(iscarbon(L))//carbon breathing
+			var/mob/living/carbon/C = L	//do it here because it happens slower than life to make atmos not die
+			if(times_fired >= C.next_breathe_check || C.failed_last_breath)
+				C.handle_breathing()
+				C.next_breathe_check = times_fired + 4
+				if(!C.failed_last_breath)//if this changes we're going to check next breathe anyway so no need to check organs
+					var/obj/item/organ/lungs/lung = C.getorganslot(ORGAN_SLOT_LUNGS)
+					var/obj/item/organ/lungs/heart = C:getorganslot(ORGAN_SLOT_HEART)
+					if(lung?.damage > lung.high_threshold)
+						C.next_breathe_check--
+					if(heart?.damage > heart.high_threshold)
+						C.next_breathe_check--
```

Comment:
Why can't this breathing chunk of code be part of life_process of carbons somehow? Feels weird to have this in the subsystem code

Reply:
because of the 

times_fired >= C.next_breathe_check

check which checks the mobs fired but i just realized i could reference it with ssmobs.times_fired

dont code tired kids

Reply:
nvm looks like tired me is smarter than i am

![image](https://user-images.githubusercontent.com/57223640/92496282-3d3fdb80-f1f8-11ea-91fe-69c9f323232c.png)

Reply:
How can it be such a difference for just changing where it's located? I'm not very familiar with the profiling tool, is this an actual benchmark or just a few samples?